uk.co.compucorp.civicrm.hrsampledata/CRM/HRSampleData/CSVHandler.php

```diff
@@ -86,6 +91,24 @@ protected function callAPI($entity, $action, $params) {
   }
 
   /**
+   * Deletes entity record from the database based on
+   * search criteria
+   *
+   * @param string $entity
+   * @param array $searchParams
+   */
+  protected function deleteRecord($entity, $searchParams) {
```

Comment:
thanks so much for the detailed explanation  , I thought the template pattern the I've used is the perfect fit for this kind or problems but apparently it appear that the visitor pattern play more nicely here , let me see how could I play it out.